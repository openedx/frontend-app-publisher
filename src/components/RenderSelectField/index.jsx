import React from 'react';
import PropTypes from 'prop-types';
import { Form } from '@edx/paragon';

const RenderSelectField = ({
  input,
  extraInput,
  name,
  label,
  disabled,
  hidden,
  required,
  meta: { touched, error },
  options,
}) => (
  <Form.Group controlId={`${name}-text-label`} isInvalid={touched && error}>
    {hidden ? null : <Form.Label>{label}</Form.Label>}
    <Form.Control
      {...input}
      {...extraInput}
      as="select"
      name={name}
      label={label}
      disabled={disabled}
      hidden={hidden}
      required={required}
    >
      {options.map(option => (
        <option key={option.value} value={option.value}>{option.label}</option>
      ))}
      {touched && error && (
      <Form.Control.Feedback>
        {error}
      </Form.Control.Feedback>
      )}
    </Form.Control>
  </Form.Group>
);

RenderSelectField.defaultProps = {
  extraInput: {},
  name: '',
  disabled: false,
  required: false,
  hidden: false,
};

RenderSelectField.propTypes = {
  input: PropTypes.shape({}).isRequired,
  extraInput: PropTypes.shape({}),
  name: PropTypes.string,
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
  meta: PropTypes.shape({
    touched: PropTypes.bool,
    error: PropTypes.string,
  }).isRequired,
  disabled: PropTypes.bool,
  required: PropTypes.bool,
  options: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.string),
    PropTypes.arrayOf(PropTypes.object),
  ]).isRequired,
  hidden: PropTypes.bool,
};

export default RenderSelectField;
