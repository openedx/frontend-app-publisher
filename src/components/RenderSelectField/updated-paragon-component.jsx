// TODO: This file can be deleted as a part of https://github.com/openedx/frontend-app-publisher/pull/761
// When we migrate off deprecated paragon components, we want to replace the contents of ./index.jsx
// with the contents of this file and then delete this file.
// Adding this file to the codebase as a temporary measure
// because the existing select component doesn't have multiselect functionality
// and it doesn't make sense to edit paragon to add that functionality bc we're moving off that component soon.

import React from 'react';
import PropTypes from 'prop-types';
import { Form } from '@edx/paragon';

const RenderSelectField = ({
  input,
  extraInput,
  name,
  label,
  disabled,
  required,
  meta: { touched, error },
  options,
}) => (
  <Form.Group controlId={`${name}-text-label`} isInvalid={touched && error}>
    <Form.Label>
      {label}
    </Form.Label>
    <Form.Control
      {...input}
      {...extraInput}
      as="select"
      name={name}
      label={label}
      disabled={disabled}
      required={required}
    >
      {options.map(option => (
        <option key={option.value} value={option.value}>{option.label}</option>
      ))}
      {touched && error && (
      <Form.Control.Feedback>
        {error}
      </Form.Control.Feedback>
      )}
    </Form.Control>
  </Form.Group>
);

RenderSelectField.defaultProps = {
  extraInput: {},
  name: '',
  disabled: false,
  required: false,
};

RenderSelectField.propTypes = {
  input: PropTypes.shape({}).isRequired,
  extraInput: PropTypes.shape({}),
  name: PropTypes.string,
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
  meta: PropTypes.shape({
    touched: PropTypes.bool,
    error: PropTypes.string,
  }).isRequired,
  disabled: PropTypes.bool,
  required: PropTypes.bool,
  options: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.string),
    PropTypes.arrayOf(PropTypes.object),
  ]).isRequired,
};

export default RenderSelectField;
