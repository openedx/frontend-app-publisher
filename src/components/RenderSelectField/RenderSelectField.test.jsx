import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import RenderSelectField from './index';

describe('RenderSelectField', () => {
  it('renders html for select field', () => {
    const component = shallow(<RenderSelectField
      input={{}}
      name="Test"
      label="TestLabel"
      options={['one', 'two', 'three']}
      meta={{ touched: false, error: '' }}
    />).dive();
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
