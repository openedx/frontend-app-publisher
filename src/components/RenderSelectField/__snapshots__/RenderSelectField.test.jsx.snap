// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderSelectField renders html for select field 1`] = `
<div
  className="pgn__form-group"
>
  <FormGroupContextProvider
    controlId="Test-text-label"
    isInvalid={false}
    isValid={false}
  >
    <FormLabel
      isInline={false}
    >
      TestLabel
    </FormLabel>
    <ForwardRef
      as="select"
      autoResize={false}
      disabled={false}
      hidden={false}
      label="TestLabel"
      name="Test"
      plaintext={false}
      required={false}
    >
      <option />
      <option />
      <option />
    </ForwardRef>
  </FormGroupContextProvider>
</div>
`;
